packages/dds/tree/src/feature-libraries/chunked-forest/codec/compressedEncode.ts

```diff
@@ -389,27 +419,80 @@ export class NestedArrayEncoder
 			outputBuffer.push(buffer);
 		}
 	}
+}
+
+/**
+ * Encodes a chunk with the {@link EncodedIncrementalChunkShape} shape.
+ * This chunks will be encoded separately, i.e., the contents of the chunk will not be part of the main buffer.
+ * A reference to the chunk will be stored in the main buffer as an {@link ChunkReferenceId}.
+ */
+export class IncrementalChunkShape extends ShapeGeneric<EncodedChunkShape> {
+	/**
+	 * Encodes all the nodes in the chunk at the cursor position using `InlineArrayShape`.
+	 */
+	public static encodeChunk(chunk: TreeChunk, context: EncoderContext): BufferFormat {
+		const chunkOutputBuffer: BufferFormat = [];
+		const nodesEncoder = asNodesEncoder(anyNodeEncoder);
+		const chunkCursor = chunk.cursor();
+		chunkCursor.firstNode();
+		const chunkLength = chunkCursor.chunkLength;
+		for (let index = 0; index < chunkLength; index++) {
+			nodesEncoder.encodeNodes(chunkCursor, context, chunkOutputBuffer);
+		}
+		assert(
+			chunkCursor.mode === CursorLocationType.Fields,
+			"should return to fields mode when finished encoding",
+		);
+		return chunkOutputBuffer;
+	}
 
 	public encodeShape(
 		identifiers: DeduplicationTable<string>,
 		shapes: DeduplicationTable<Shape>,
 	): EncodedChunkShape {
-		const shape: EncodedNestedArrayShape =
-			shapes.valueToIndex.get(this.inner.shape) ??
-			fail(0xb4f /* index for shape not found in table */);
 		return {
-			a: shape,
+			e: 0 /* EncodedIncrementalChunkShape */,
 		};
 	}
 
 	public countReferencedShapesAndIdentifiers(
 		identifiers: Counter<string>,
 		shapeDiscovered: (shape: Shape) => void,
-	): void {
-		shapeDiscovered(this.inner.shape);
+	): void {}
+
+	public get shape(): this {
+		return this;
 	}
 }
 
+/**
+ * Encodes an incremental field whose chunks are encoded separately and referenced by their {@link ChunkReferenceId}.
+ * The shape of the content of this field is {@link NestedShape} where the items in the array are
+ * the {@link ChunkReferenceId}s of the encoded chunks.
+ */
+export const incrementalFieldEncoder: FieldEncoder = {
+	encodeField(
+		cursor: ITreeCursorSynchronous,
+		context: EncoderContext,
+		outputBuffer: BufferFormat,
+	): void {
+		assert(
+			context.shouldEncodeIncrementally,
+			"incremental encoding must be enabled to use IncrementalFieldShape",
+		);
+
+		let chunkReferenceIds: ChunkReferenceId[] = [];
+		if (cursor.getFieldLength() !== 0) {
```

Comment:
IncrementalEncoder.encodeIncrementalField does not say anything about not supporting empty fields so I think we should skip this check, and let it handle the empty case. I don't think we gain anything by adding the special casing of empty here.

Reply:
I added this check so that we don't end up adding an empty summary tree for fields that are empty. I could move this check to the `ForestIncrementalSummaryBuilder` and it can decide to return an empty array in case the field is empty.